lib/matplotlib/font_manager.py

```diff
@@ -1273,36 +1270,34 @@ def _findfont_cached(self, prop, fontext, directory, fallback_to_default,
             if score == 0:
                 break
 
-        if best_font is None or best_score >= 10.0:
-            if fallback_to_default:
-                _log.warning(
-                    'findfont: Font family %s not found. Falling back to %s.',
-                    prop.get_family(), self.defaultFamily[fontext])
-                default_prop = prop.copy()
-                default_prop.set_family(self.defaultFamily[fontext])
-                return self.findfont(default_prop, fontext, directory, False)
-            else:
-                # This is a hard fail -- we can't find anything reasonable,
-                # so just return the DejaVuSans.ttf
-                _log.warning('findfont: Could not match %s. Returning %s.',
-                             prop, self.defaultFont[fontext])
-                result = self.defaultFont[fontext]
-        else:
+        if (best_font is not None and os.path.isfile( best_font.fname)
+                and best_score < 10):
             _log.debug('findfont: Matching %s to %s (%r) with score of %f.',
                        prop, best_font.name, best_font.fname, best_score)
-            result = best_font.fname
-
-        if not os.path.isfile(result):
-            if rebuild_if_missing:
-                _log.info(
-                    'findfont: Found a missing font file.  Rebuilding cache.')
-                _rebuild()
-                return fontManager.findfont(
-                    prop, fontext, directory, True, False)
-            else:
-                raise ValueError("No valid font could be found")
-
-        return result
+            return best_font.fname
+
+        # No sufficently good font found.
+        if rebuild_if_missing:
+            _log.info(
+                'findfont: Found a missing font file.  Rebuilding cache.')
+            _rebuild()
+            return self.findfont(prop, fontext, directory, True, False)
```

Comment:
please pass True and False as keyword arguments (too hard to know what they mean otherwise) (yes i know the old code was as bad)